### DISCUSSION TITLE
Global PubSub

### DISCUSSION THREAD

**Discussion by Sytten** [tqwewe/kameo#89] [General]
Hi there!
Our system is currently on Actix & Actix-broker. Slowly looking to move to a better system.
Right now a couple of our actors do something like that on start:
```rust
fn started(&mut self, ctx: &mut Self::Context) {
            use actix_broker::BrokerSubscribe;
            self.subscribe_system_async::<MyMessage>(ctx);
}
```
Essentially I would need a global PubSub actor right? I can likely spawn one and store its ref in a static global variable.
Any thoughts?

**Comment by tqwewe** [comment]
The only solution kameo provides would be sharing a global PubSub actor across your system, possibly with a static global variable as you mentioned.

If you need more flexibility, kameo lacks offering much more than that sadly. But there's a discussion here with might be relevant in case you need to support many types for the global pubsub actor: https://github.com/tqwewe/kameo/discussions/83#discussioncomment-11298230

**Reply by Sytten** [reply]
Thanks for the reply!
